packages/theme/src/gui-provider.tsx

```diff
@@ -23,18 +38,33 @@ GuiContext.displayName = 'GuiContext';
 export interface GuiProviderProps {
   children: React.ReactNode;
   theme?: GuiTheme;
-  colorScheme?: string;
+  /**
+   * @default system
+   */
+  colorScheme?: ColorSchemeMode;
 }
 
 export function GuiProvider({ theme = defaultTheme, colorScheme: _colorScheme, children }: GuiProviderProps) {
-  const [preferredColorScheme, setPreferredColorScheme] = React.useState<'light' | 'dark'>('light');
+  const [preferredColorScheme, setPreferredColorScheme] = React.useState<
+    Exclude<ColorSchemeMode, ColorSchemeMode.System>
+  >(ColorSchemeMode.Light);
+  const [pickedColorScheme, setPickedColorScheme] = React.useState<ColorSchemeMode>(ColorSchemeMode.System);
+
+  React.useEffect(() => {
+    setPickedColorScheme(_colorScheme || ColorSchemeMode.System);
+  }, [_colorScheme]);
 
   useSafeEffect(() => {
-    const scheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
-    setPreferredColorScheme(scheme);
-  }, []);
+    if (pickedColorScheme === ColorSchemeMode.System) {
+      const scheme = window.matchMedia('(prefers-color-scheme: dark)').matches
+        ? ColorSchemeMode.Dark
+        : ColorSchemeMode.Light;
+      setPreferredColorScheme(scheme);
+    }
+  }, [pickedColorScheme]);
 
-  const colorScheme = _colorScheme || preferredColorScheme;
+  const colorScheme =
+    pickedColorScheme === ColorSchemeMode.System ? preferredColorScheme : pickedColorScheme || preferredColorScheme;

```

Comment:
Shouldn't this effectively be the same as:
const colorScheme =
    pickedColorScheme === ColorSchemeMode.System ? preferredColorScheme : pickedColorScheme
    
Because pickedColorScheme has a value on init? Or is this a fallback for when a user enters an invalid value? I might be missing something

Reply:
You are right. While developing, undefined could still be used.